SPEC.md

```diff
@@ -140,4 +140,127 @@ When creating a new sample, follow these steps to apply the ruleset to the sampl
 ### Linting
 
 All samples must have the following linting configuration enabled.
-<TBD>
\ No newline at end of file
+<TBD>
+
+## Bot Styles
+We want to have consistent patterns for how to use our SDK.  The following should be followed for all of our samples.
+
+### General Rules
+
+#### Startup initialization should only initialize middleware and bot
+
+> We want to minimize the cognitive load of what happens in startup and 
+> foster code being organized along class boundaries.
+
+#### Creating Property Accessors should be scoped to the class which makes the most sense.
+
+> We want to organize our logic at class boundaries to make it clearer 
+> how code reuse happens and localize the information around classes.
+
+```cs
+    public class ExampleDialog
+    {
+        public ExampleDialog(BotState state)
+        {
+            this.AlarmsProperty = state.CreateProperty<Alarm[]>("Alarms");
+        }
+    	IStatePropertyAccessor<Alarms> AlarmsProperty { get;set;}
+    }
+    ...
+    var alarms = await this.AlarmsProperty.Get(context);
+```
+
+#### Property Accessors should be passed to a class if they are shared among multiple classes
+
+> If you have 3 dialogs which all need the same property accessor, then the higher level
+> class should define the accessor and pass it to the child dialogs. 
+
+```cs
+   AlarmDialogs(BotState botState)
+   {
+       this.AlarmsProperty = botState.CreateProperty<List<Alarm>>("Alarms");
+
+       this.Dialogs.Add(new AddAlarmDialog(alarmsDialog));
+       this.Dialogs.Add(new ShowAlarmsDialog(alarmsDialog));
+       this.Dialogs.Add(new DeleteAlarmsDialog(alarmsDialog));
+   } 
+   
+```
+
+#### Constructors should use BotState to create properties unless the component requires that it is ConversationState or UserState.
+
+> Reasoning:  When you take a ConversationState you are stating that your component is
+> required to be have the scoping of conversation for it to work correctly.  When you 
+> accept BotState as your input parameter you are leaving your class open to have your 
+> state live where the caller needs it to be, and makes you work with future storage 
+> types. 
+
+```cs
+    public class ExampleDialog
+    {
+        public ExampleDialog(BotState state)
+        {
+            this.AlarmsProperty = state.CreateProperty<Alarm[]>("Alarms");
+        }
+```
+
```

Comment:
@tomlm - I do not follow this one. If I use BotState for Alarms but in reality Alarms should be tied to a specific user, I'd have expected to see Alarms under userState. 

I also dont yet know the kind of information I'd want to keep in Bot State - what pieces of data relevant to my bot do you believe need to live beyond the scope of the current conversation or user? 

May be a different example that demonstrates why and what I'd want to keep in each state scopes (Bot/ User/ Conversation) would be really helpful.